### PR TITLE
Don't double install mac deps, cleanups

### PR DESCRIPTION
**Pull request by ankith26**
This PR is one part of the series of changes I'm making. The goal of this PR is to update the scripts to only install mac deps in the custom prefix (and not in `/usr/local`) and do related cleanups that should eventually help us get on using self-compiled deps on windows, and also being able to use systems like meson for the dependencies that either only support it, or recommend using it

### CODE REVIEW

**Review by MyreMylar — APPROVED** [review]
LGTM and, more importantly here, the mac CI is happy.

File: buildconfig/manylinux-build/docker_base/opus/build-opus.sh [hunk 1]
```diff
@@ -13,25 +13,15 @@ sha512sum -c opus.sha512
...
-
 cd ..
 
 tar xzf ${OPUS_FILE}.tar.gz
 cd $OPUS_FILE
 
-./configure $ARCHS_CONFIG_FLAG --disable-http
+./configure $PG_BASE_CONFIGURE_FLAGS --disable-http
 make
 make install
```
Comment (Starbuck5) [inline buildconfig/manylinux-build/docker_base/opus/build-opus.sh:27]: GH diff says this file is lacking a trailing newline

**Comment by Starbuck5** [comment]
Question: does this lead to a noticeable speedup of the mac deps generation? That's not necessary, but it would be a quantifiable win for this PR.

> also being able to use systems like meson

I used Meson for a C/C++ project last year and I had a good impression of it. Other libraries seem to be getting good use of it, I think numpy and scipy use meson now.

**Comment by Starbuck5** [comment]
@oddbookworm or @novialriptide Would it be convenient for either of you to give these mac wheels a test?

**Comment by ankith26** [comment]
> Question: does this lead to a noticeable speedup of the mac deps generation? 

In theory it should, but the speedup would be small. There is usually a lot of randomness in the time the build step takes, for some reason. But it runs rarely so I guess it won't be a big deal.

File: buildconfig/manylinux-build/docker_base/opus/build-opus.sh (hunk 1 above)
Comment (ankith26) [reply buildconfig/manylinux-build/docker_base/opus/build-opus.sh:27]: I know it looks like it, but actually doesn't. What the diff means here is that the line missing the trailing newline was deleted

**Comment by novialriptide** [comment]
I got multiple errors.

<details close>
<summary>click to open logs (`ankith26-mac-build` branch)</summary>
```
======================================================================
ERROR: test_get__owned_empty_type (pygame.tests.scrap_test.ScrapModuleTest.test_get__owned_empty_type)
Ensures get works when there is no data of the requested type
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/opt/homebrew/lib/python3.11/site-packages/pygame/tests/scrap_test.py", line 64, in test_get__owned_empty_type
    scrap.put(pygame.SCRAP_TEXT, b"text to clipboard")
pygame.error: content could not be placed in clipboard.

======================================================================
ERROR: test_init__reinit (pygame.tests.scrap_test.ScrapModuleTest.test_init__reinit)
Ensures reinitializing the scrap module doesn't clear its data.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/opt/homebrew/lib/python3.11/site-packages/pygame/tests/scrap_test.py", line 36, in test_init__reinit
    scrap.put(data_type, expected_data)
pygame.error: content could not be placed in clipboard.

======================================================================
ERROR: test_put (pygame.tests.scrap_test.ScrapModuleTest.test_put)
Ensures put can place data into the clipboard
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/opt/homebrew/lib/python3.11/site-packages/pygame/tests/scrap_test.py", line 113, in test_put
    scrap.put(DATA_TYPE, b"buf")
pygame.error: content could not be placed in clipboard.

======================================================================
ERROR: test_put__bmp_image (pygame.tests.scrap_test.ScrapModuleTest.test_put__bmp_image)
Ensures put can place a BMP image into the clipboard.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/opt/homebrew/lib/python3.11/site-packages/pygame/tests/scrap_test.py", line 103, in test_put__bmp_image
    scrap.put(pygame.SCRAP_BMP, expected_bytes)
pygame.error: content could not be placed in clipboard.

======================================================================
ERROR: test_put__text (pygame.tests.scrap_test.ScrapModuleTest.test_put__text)
Ensures put can place text into the clipboard.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/opt/homebrew/lib/python3.11/site-packages/pygame/tests/scrap_test.py", line 90, in test_put__text
    scrap.put(pygame.SCRAP_TEXT, b"Hello world")
pygame.error: content could not be placed in clipboard.

======================================================================
FAIL: test_multiple_timers (pygame.tests.time_test.TimeModuleTest.test_multiple_timers)
Test for multiple timers running together, test repeat and dict
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/opt/homebrew/lib/python3.11/site-packages/pygame/tests/time_test.py", line 287, in test_multiple_timers
    self.assertAlmostEqual(
AssertionError: 14.290542007074691 != 10 within 4 delta (4.290542007074691 difference)

======================================================================
FAIL: test_box_blur (pygame.tests.transform_test.TransformDisplayModuleTest.test_box_blur)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/opt/homebrew/lib/python3.11/site-packages/pygame/tests/transform_test.py", line 1501, in test_box_blur
    self.assertTrue(sf_b2.get_at(pos) == data2[pos])
AssertionError: False is not true

======================================================================
FAIL: test_gaussian_blur (pygame.tests.transform_test.TransformDisplayModuleTest.test_gaussian_blur)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/opt/homebrew/lib/python3.11/site-packages/pygame/tests/transform_test.py", line 1543, in test_gaussian_blur
    self.assertTrue(sf_b1.get_at(pos) == data1[pos])
AssertionError: False is not true

======================================================================
FAIL: test_position (pygame.tests.window_test.WindowTypeTest.test_position)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/opt/homebrew/lib/python3.11/site-packages/pygame/tests/window_test.py", line 130, in test_position
    self.assertTupleEqual(self.win.position, (12, 34))
AssertionError: Tuples differ: (12, 72) != (12, 34)

First differing element 1:
72
34

- (12, 72)
?      ^^

+ (12, 34)
?      ^^


----------------------------------------------------------------------
Ran 2095 tests in 47.046s

FAILED (failures=4, errors=5)
```
</details>


<details close>
<summary>click to open logs (version `2.3.2`)</summary>
```
======================================================================
ERROR: test_get__owned_empty_type (pygame.tests.scrap_test.ScrapModuleTest.test_get__owned_empty_type)
Ensures get works when there is no data of the requested type
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/opt/homebrew/lib/python3.11/site-packages/pygame/tests/scrap_test.py", line 64, in test_get__owned_empty_type
    scrap.put(pygame.SCRAP_TEXT, b"text to clipboard")
pygame.error: content could not be placed in clipboard.

======================================================================
ERROR: test_init__reinit (pygame.tests.scrap_test.ScrapModuleTest.test_init__reinit)
Ensures reinitializing the scrap module doesn't clear its data.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/opt/homebrew/lib/python3.11/site-packages/pygame/tests/scrap_test.py", line 36, in test_init__reinit
    scrap.put(data_type, expected_data)
pygame.error: content could not be placed in clipboard.

======================================================================
ERROR: test_put (pygame.tests.scrap_test.ScrapModuleTest.test_put)
Ensures put can place data into the clipboard
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/opt/homebrew/lib/python3.11/site-packages/pygame/tests/scrap_test.py", line 113, in test_put
    scrap.put(DATA_TYPE, b"buf")
pygame.error: content could not be placed in clipboard.

======================================================================
ERROR: test_put__bmp_image (pygame.tests.scrap_test.ScrapModuleTest.test_put__bmp_image)
Ensures put can place a BMP image into the clipboard.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/opt/homebrew/lib/python3.11/site-packages/pygame/tests/scrap_test.py", line 103, in test_put__bmp_image
    scrap.put(pygame.SCRAP_BMP, expected_bytes)
pygame.error: content could not be placed in clipboard.

======================================================================
ERROR: test_put__text (pygame.tests.scrap_test.ScrapModuleTest.test_put__text)
Ensures put can place text into the clipboard.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/opt/homebrew/lib/python3.11/site-packages/pygame/tests/scrap_test.py", line 90, in test_put__text
    scrap.put(pygame.SCRAP_TEXT, b"Hello world")
pygame.error: content could not be placed in clipboard.

======================================================================
FAIL: test_multiple_timers (pygame.tests.time_test.TimeModuleTest.test_multiple_timers)
Test for multiple timers running together, test repeat and dict
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/opt/homebrew/lib/python3.11/site-packages/pygame/tests/time_test.py", line 287, in test_multiple_timers
    self.assertAlmostEqual(
AssertionError: 43.70808300154749 != 10 within 4 delta (33.70808300154749 difference)

======================================================================
FAIL: test_box_blur (pygame.tests.transform_test.TransformDisplayModuleTest.test_box_blur)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/opt/homebrew/lib/python3.11/site-packages/pygame/tests/transform_test.py", line 1501, in test_box_blur
    self.assertTrue(sf_b2.get_at(pos) == data2[pos])
AssertionError: False is not true

======================================================================
FAIL: test_gaussian_blur (pygame.tests.transform_test.TransformDisplayModuleTest.test_gaussian_blur)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/opt/homebrew/lib/python3.11/site-packages/pygame/tests/transform_test.py", line 1543, in test_gaussian_blur
    self.assertTrue(sf_b1.get_at(pos) == data1[pos])
AssertionError: False is not true

======================================================================
FAIL: test_pg_version_consistency (pygame.tests.version_test.VersionTest.test_pg_version_consistency)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/opt/homebrew/lib/python3.11/site-packages/pygame/tests/version_test.py", line 38, in test_pg_version_consistency
    self.assertEqual(pgh_minor, version.vernum[1])
AssertionError: 4 != 3

----------------------------------------------------------------------
Ran 2046 tests in 40.783s

FAILED (failures=4, errors=5)
```
</details>

**Comment by novialriptide** [comment]
It could be because I'm using ARM (Apple Silicon), not x86 (Intel)?

**Comment by ankith26** [comment]
I get those scrap fails locally too so that's fine IG. What's weird is the timer fail and the blur fail (but again, both happen on the older version too). I don't think either is caused by this PR though. Could you also go though a few examples and stuff to check those are not broken

**Comment by novialriptide** [comment]
I did a quick run of the following
 - `aliens.py`
 - `go_over_there.py`
 - `liquid.py` (this was pretty cool)
 - `stars.py`

Nothing seems broken so far.

**Pull request merged**